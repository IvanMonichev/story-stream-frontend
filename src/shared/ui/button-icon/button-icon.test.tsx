import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { describe, expect, it, vi } from 'vitest'
import ButtonIcon from './button-icon'

vi.mock('@/shared/core/icon/icon', () => ({
  default: ({ icon }: { icon: string }) => (
    <span data-testid='icon'>{icon}</span>
  )
}))

describe('ButtonIcon component', () => {
  it('renders the icon', () => {
    render(<ButtonIcon icon='search' />)
    expect(screen.getByTestId('icon')).toHaveTextContent('search')
  })

  it('renders children correctly', () => {
    render(<ButtonIcon icon='plus'>Add</ButtonIcon>)
    expect(screen.getByRole('button')).toHaveTextContent('Add')
  })

  it('applies custom className', () => {
    render(<ButtonIcon icon='check' className='custom-class' />)
    expect(screen.getByRole('button')).toHaveClass('custom-class')
  })

  it('passes additional button props', async () => {
    const handleClick = vi.fn()
    render(
      <ButtonIcon icon='save' type='submit' onClick={handleClick}>
        Save
      </ButtonIcon>
    )
    const button = screen.getByRole('button')
    expect(button).toHaveAttribute('type', 'submit')
    await userEvent.click(button)
    expect(handleClick).toHaveBeenCalled()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ButtonIcon icon='download'>Download</ButtonIcon>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
