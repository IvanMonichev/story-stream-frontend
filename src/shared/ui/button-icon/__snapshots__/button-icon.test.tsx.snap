// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonIcon component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="_button_c80904"
  >
    <span
      data-testid="icon"
    >
      download
    </span>
    Download
  </button>
</DocumentFragment>
`;
